### PR TITLE
Add support for former editors

### LINKED ISSUE #1097: Add support for former editors
A bit superfluous, but in some cases the editor section can be a bit of a mess. For example, here is the APG’s editor listing:

List of 7 APG editors including 3 previous editors. It shows editor’s name, company, and e-mail

ARIA’s editor listing is a bit better, but that’s mainly because their company names are shorter.

List of 4 ARIA editors. It shows editor’s name, company, and e-mail.

To compare that with the cleaner look of the HTML spec editor listing:

Separate lists for 5 current HTML editors and 3 former HTML editors. It lists their name (which is linked to either their email or website) and their company name between brackets.

Thoughts about moving towards the HTML look?

### PR DESCRIPTION
**Pull request by sidvishnoi**
Make editor section cleaner. Fixes #1097

I've made a new `conf` attribute called `formerEditors`. It is same as `editors`, but shown in different section.
Alternative: add an optional attribute (like `isFormer: bool` to `People`) and filter based on it (but I think adding a separate variable for former editors is better, keeps things at one place)

TODO: i10n

### CODE REVIEW

**Comment by marcoscaceres** [comment]
I think we should go with `formerEditors`. I’ll try to review soon.

**Comment by saschanaz** [comment]
Side note: You may want to add "Fixes #1097" on the PR description instead of title, as it enables automatic linking and issue closing.

File: src/w3c/headers.js [hunk 1]
```diff
@@ -377,10 +378,14 @@ export function run(conf, doc, cb) {
...
     conf.editors.forEach(peopCheck);
   }
+  if (conf.formerEditors) {
+    conf.formerEditors.forEach(peopCheck);
+  }
   if (conf.authors) {
     conf.authors.forEach(peopCheck);
   }
   conf.multipleEditors = conf.editors && conf.editors.length > 1;
+  conf.multipleFormerEditors = conf.formerEditors && conf.formerEditors.length > 1;
```
Comment (marcoscaceres) [inline src/w3c/headers.js]: Maybe: 
```JS
conf.multipleFormerEditors = Array.isArray(conf.formerEditors) && conf.formerEditors.length;
```

File: src/w3c/templates/cgbg-headers.js [hunk 2]
```diff
@@ -49,6 +49,10 @@ export default conf => {
     ` : ""}
     <dt>${conf.multipleEditors ? html`${conf.l10n.editors}` : html`${conf.l10n.editor}`}</dt>
     ${showPeople(conf, "Editor", conf.editors)}
+    ${conf.formerEditors ? html`
+      <dt>${conf.multipleFormerEditors ? html`${conf.l10n.former_editors}` : html`${conf.l10n.former_editor}`}</dt>
```
Comment (marcoscaceres) [inline src/w3c/templates/cgbg-headers.js]: nit:
```HTML
<dt>${conf.multipleFormerEditors ? conf.l10n.former_editors : conf.l10n.former_editor}</dt>
```
These are always a string, not HTML.

File: src/w3c/templates/headers.js [hunk 3]
```diff
@@ -56,6 +56,10 @@ export default conf => {
     ` : ""}
     <dt>${conf.multipleEditors ? html`${conf.l10n.editors}` : html`${conf.l10n.editor}`}</dt>
     ${showPeople(conf, "Editor", conf.editors)}
+    ${conf.formerEditors ? html`
+      <dt>${conf.multipleFormerEditors ? html`${conf.l10n.former_editors}` : html`${conf.l10n.former_editor}`}</dt>
```
Comment (marcoscaceres) [inline src/w3c/templates/headers.js]: As above.

File: src/w3c/templates/show-people.js [hunk 4]
```diff
@@ -12,23 +12,22 @@ export default (conf, name, items = []) => {
...
     } else {
       contents.push(html`<span class='p-name fn'>${p.name}</span>`);
     }
     if (p.company) {
       if (p.companyURL) {
-        contents.push(html`, <a class='p-org org h-org h-card' href='${p.companyURL}'>${p.company}</a>`);
+        contents.push(html` (<a class='p-org org h-org h-card' href='${p.companyURL}'>${p.company}</a>)`);
       } else {
-        contents.push(document.createTextNode(`, ${p.company}`));
+        contents.push(document.createTextNode(` (${p.company})`));
```
Comment (marcoscaceres) [inline src/w3c/templates/show-people.js]: Would "html` (${p.company})`" not do the same thing (i.e. create the text node)?

File: src/w3c/templates/cgbg-headers.js (hunk 2 above)
Comment (sidvishnoi) [reply src/w3c/templates/cgbg-headers.js]: did that based on code above (like for `multipleEditors`). will fix them also.

**Comment by sidvishnoi** [comment]
I've made the fixes as you've mentioned

The tests I've added are primarily based on tests of "editors"

File: tests/spec/w3c/headers-spec.js [hunk 5]
```diff
@@ -179,6 +179,88 @@ describe("W3C — Headers", function() {
...
+          companyURL: "http://COMPANY",
+          mailto: "EMAIL",
+          w3cid: "1234",
+        }],
+      };
+      Object.assign(ops.config, newProps);
+      const doc = await makeRSDoc(ops);
+
+      expect($("dt:contains('Former Editors:')", doc).length).toEqual(0);
+
```
Comment (marcoscaceres) [inline tests/spec/w3c/headers-spec.js]: Sorry 😐 we are trying not to use jQuery in the project any more. Could you change these to just use querySelectorAll().

Comment (sidvishnoi) [reply tests/spec/w3c/headers-spec.js]: Oh. I thought we could use jQuery in tests.
No problem. I'll update this.

Comment (marcoscaceres) [reply tests/spec/w3c/headers-spec.js]: It actually turns out to be bad in tests too, as it adds a layer of indirection that makes things harder to debug when things go wrong.

Comment (sidvishnoi) [reply tests/spec/w3c/headers-spec.js]: I'm thinking of adding these helper functions to `specHelper.js`: https://gist.github.com/sidvishnoi/2614f24af7d11051451a6a244bcecefb

Please do comment on that.

Comment (marcoscaceres) [reply tests/spec/w3c/headers-spec.js]: Sorry, I’m not clear which helper functions you mean. Could you clarify?

Comment (sidvishnoi) [reply tests/spec/w3c/headers-spec.js]: As javascript doesn't have one-liners for selectors like `dt:contains(someText)` and `el.next("dd")`, I'm writing the helper functions

For example, `selectorContains(selector, text, context = document)` equivalent to jQuery's `:contains` selector

I've added them in gist above.

Unless we plan to use some other selector library like Sizzle or create all the helper functions as we need them.

Comment (marcoscaceres) [reply tests/spec/w3c/headers-spec.js]: I could be convinced, but I’ve personally not found a need for them.

Comment (sidvishnoi) [reply tests/spec/w3c/headers-spec.js]: Can you suggest alternative to the following? 
``` js
expect($("dt:contains('Former Editors:')", doc).length).toEqual(0);
```

Comment (marcoscaceres) [reply tests/spec/w3c/headers-spec.js]: So, a proposal elsewhere was to bring in Sizzle as a starting point. I’d be open to that - but would only want us to use it on the rare case the web platform can’t do something (e.g. “contains”)

Comment (sidvishnoi) [reply tests/spec/w3c/headers-spec.js]: In the test files, I've seen a huge usage of `:contains` and `el.next(tagname)`. Most of selectors are supported with `querySelector[All]`. 

So what do you suggest for the test cases for this module? Should I continue to use jQuery like selectors until we finalise on this? Or create helpers?

Comment (sidvishnoi) [reply tests/spec/w3c/headers-spec.js]: I will create a list of helper functions we may need that can help in making a decision whether to use some library like Sizzle or create our own (if a few).

Comment (marcoscaceres) [reply tests/spec/w3c/headers-spec.js]: Sounds great!

Comment (marcoscaceres) [reply tests/spec/w3c/headers-spec.js]: About suggestion from above:

const eds= [...doc.querySelectorAll(“dt”)].find({textContent}) => textContent.trim() === “Former Editors:”);

Or some variation of the above.

Comment (sidvishnoi) [reply tests/spec/w3c/headers-spec.js]: gets a little verbose and repetitive though

Comment (marcoscaceres) [reply tests/spec/w3c/headers-spec.js]: Yes, sometimes it definitely does. For the tests at least, the most important thing is limiting indirection (even at the cost of some repetition). But we make exceptions, like the makeRSDoc() function.

**Comment by marcoscaceres** [comment]
@sidvishnoi, I had misunderstood what you mean by repetition with the lookup. I think it's totally fine to make helper functions in the tests, so long as they are super light weight. I've added a generic content lookup there for now. I also cleaned up a few things. Please take a look!

**Comment by sidvishnoi** [comment]
This looks a lot better! Thank you!

Just one question. Instead of doing `[...dtElems]` multiple times, wouldn't it be better to do `const dtElems = [... doc.querySelectorAll("dt")];` once?

**Comment by marcoscaceres** [comment]
Absolutely, do it! 👌

**Comment by sidvishnoi** [comment]
> document this in the wiki

on it :smile:

**Pull request merged**